File: Models/Transaction.cs
namespace financial_manager.Models
{
    public class Transaction
    {
        public int Id { get; init; }
        public string Title { get; set; } = string.Empty;
        public decimal Significance { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? ExpenseDate { get; set; } = DateTime.Now;
        public Category? Category { get; set; }

        private string _transactionType;
        public string TransactionType
        {
            get => _transactionType;
            set
            {
                if (value != "income" && value != "expense")
                {
                    throw new ArgumentException("Transaction type must be either 'income' or 'expense'.");
                }
                _transactionType = value;
            }
        }

        public Transaction() { }
    }
}

File: financialManagerUI/src/app/entities/models/transaction.ts
import { BaseEntity } from '../shared/baseEntity';
import { Category } from './category';

export class Transaction extends BaseEntity {
  public title: string;
  public significance: number;
  public transactionType: string;
  public createdAt: Date = new Date();
  public expenseDate?: Date;
  public category?: Category = undefined;

  constructor(
    title: string,
    significance: number,
    transactionType: string,
    expenseDate: Date = new Date(),
    category?: Category,
    id?: number
  ) {
    super();
    this.title = title;
    this.significance = significance;
    this.transactionType = transactionType;
    this.expenseDate = expenseDate;
    this.category = category;
    this.id = id;
  }
}
